ServerCore/ModelBases/EventSpecificPageModel.cs

```diff
@@ -29,25 +30,35 @@ public PuzzleUser LoggedInUser
             {
                 if (loggedInUser == null)
                 {
-                    loggedInUser = PuzzleUser.GetPuzzleUserForCurrentUser(puzzleServerContext, User, userManager);
+                    loggedInUser = PuzzleUser.GetPuzzleUserForCurrentUser(_context, User, userManager);
                 }
 
                 return loggedInUser;
             }
         }
 
-        private PuzzleServerContext puzzleServerContext;
+        protected readonly PuzzleServerContext _context;
         private UserManager<IdentityUser> userManager;
 
-        public EventSpecificPageModel()
+        public EventSpecificPageModel(PuzzleServerContext serverContext, UserManager<IdentityUser> manager)
         {
-            // Default constructor - note that pages that use this constructor won't know what PuzzleUser is signed in
+            _context = serverContext;
+            userManager = manager;
         }
 
-        public EventSpecificPageModel(PuzzleServerContext serverContext, UserManager<IdentityUser> manager)
+        public async Task<bool> CanAdminPuzzle(Puzzle puzzle)
```

Comment:
This should be handled through the authorization policy. Give me a little time now that I'm not visiting family and I'll get those changes merged in.

Reply:
Thanks for checking in the authorization stuff - which parts of this can the authorization policy handle?